src/browser/api_protocol/api_handlers/system.js

```diff
@@ -78,386 +73,303 @@ function SystemApiHandler() {
     }
 
-    function raiseEvent(identity, message, ack) {
-        let evt = message.payload.eventName;
-        let eventArgs = message.payload.eventArgs;
-
-        System.raiseEvent(evt, eventArgs);
-        ack(successAck);
+    function raiseEvent(identity, message) {
+        const { payload: { eventName, eventArgs } } = message;
+        System.raiseEvent(eventName, eventArgs);
     }
 
-    function getElIPCConfig(identity, message, ack) {
-        let dataAck = _.clone(successAck);
-
-        dataAck.data = System.getElIPCConfiguration();
-        ack(dataAck);
+    function getElIPCConfig(identity, message) {
+        return System.getElIPCConfiguration();
     }
 
-    function convertOptions(identity, message, ack) {
-        let dataAck = _.clone(successAck);
-
-        dataAck.data = System.convertOptions(message.payload);
-        ack(dataAck);
-
+    function convertOptions(identity, message) {
+        return System.convertOptions(message.payload);
     }
 
-    function getWebSocketState(identity, message, ack) {
-        let dataAck = _.clone(successAck);
-
-        dataAck.data = System.getWebSocketServerState();
-        ack(dataAck);
+    function getWebSocketState(identity, message) {
+        return System.getWebSocketServerState();
     }
 
-    function generateGuid(identity, message, ack) {
-        let dataAck = _.clone(successAck);
-        dataAck.data = System.generateGUID();
-        ack(dataAck);
+    function generateGuid(identity, message) {
+        return System.generateGUID();
     }
 
-    function showDeveloperTools(identity, message, ack) {
+    function showDeveloperTools(identity, message) {
         System.showDeveloperTools(message.payload.uuid, message.payload.name);
-        ack(successAck);
     }
 
-    function clearCache(identity, message, ack, nack) {
-        System.clearCache(message.payload, (err) => {
-            if (!err) {
-                ack(successAck);
-            } else {
-                nack(err);
-            }
+    function clearCache(identity, message) {
+        return new Promise((resolve, reject) => {
+            System.clearCache(message.payload, err => {
+                if (!err) {
+                    resolve();
+                } else {
+                    reject(err);
+                }
+            });
         });
     }
 
-    function deleteCacheRequest(identity, message, ack, nack) {
+    function deleteCacheRequest(identity, message) {
         // deleteCacheOnRestart has been deprecated; redirects
         // to deleteCacheOnExit
-        System.deleteCacheOnExit(() => {
-            ack(successAck);
-        }, nack);
+
+        return new Promise((resolve, reject) => {
+            System.deleteCacheOnExit(resolve, reject);
+        });
     }
 
-    function exitDesktop(identity, message, ack) {
-        ack(successAck);
-        System.exit();
+    function exitDesktop(identity, message) {
+        setTimeout(() => System.exit());
```

Comment:
Why is this timed out here? Comment? Test?

Reply:
I'm forcing the actual exit call to come _after_ the ack per original